### PR TITLE
rewrite application update method

### PR DESCRIPTION
**Pull request by pwzgorilla**

### CODE REVIEW

File: backend/update.go [hunk 1]
```diff
@@ -192,3 +200,39 @@ func (b *Backend) doUpdate(tasks []*types.Task, version *types.Version) error {
...
+				failureTimes++
+			}
+			if conn != nil {
+				conn.Close()
+			}
+		case <-quit:
+			ticker.Stop()
+			return nil
+		}
+	}
```
Comment (upccup) [inline backend/update.go]: why there not have check success and return nil

File: backend/rollback.go [hunk 2]
```diff
@@ -42,11 +42,31 @@ func (b *Backend) RollbackApplication(applicationId string) error {
 		return err
 	}
 
-	tasks, err := b.store.ListTasks(applicationId)
+	tasks, err := b.store.ListTasks(appId)
 	if err != nil {
 		return err
 	}
 
+	go func() error {
```
Comment (upccup) [inline backend/rollback.go]: return value was not used

File: backend/update.go (hunk 1 above)
Comment (pwzgorilla) [inline backend/update.go]: cause health check is finished and no error occured

Comment (pwzgorilla) [inline backend/update.go]: @upccup cause health check is finished and no error occured

Comment (upccup) [inline backend/update.go]: @pwzgorilla it's mean all checks must wait time of UpdateDelay?

Comment (pwzgorilla) [inline backend/update.go]: @upccup yes :)

File: backend/rollback.go (hunk 2 above)
Comment (pwzgorilla) [inline backend/rollback.go]: @upccup yes, i haven't known how to use it , i'm designing. it will be used in future.

**Pull request merged**